src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -218,12 +219,17 @@ class DateColumnType(val time: Boolean): ColumnType() {
         }
 
         return if (time)
-            "'${DEFAULT_DATE_TIME_STRING_FORMATTER.print(dateTime.toDateTime(DateTimeZone.getDefault()))}'"
+            if (withTimezone) {
+                "'${DEFAULT_DATE_TIME_ZONE_STRING_FORMATTER.print(dateTime)}'"
+            } else {
+                "'${DEFAULT_DATE_TIME_STRING_FORMATTER.print(dateTime.toDateTime(DateTimeZone.getDefault()))}'"
+            }
         else
             "'${DEFAULT_DATE_STRING_FORMATTER.print(dateTime)}'"
     }
 
     override fun valueFromDB(value: Any): Any = when(value) {
+        is DateTime -> value
```

Comment:
Looks like [upstream has changed this as well](https://github.com/JetBrains/Exposed/blob/master/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt#L227)